arches_project/arches_project.py

```diff
@@ -518,47 +498,75 @@ def edit_resource(self, replace):
                                            arches_selected_resource=self.arches_selected_resource,                                            
                                            dlg=self.dlg,
                                            dlg_edit_resource_replace=self.dlg_edit_resource_replace, 
-                                           dlg_edit_resource_add=self.dlg_edit_resource_add)
-
-
+                                           dlg_edit_resource_add=self.dlg_edit_resource_add,
+                                           iface=self.iface)
 
     def arches_connection_save(self):
         """
         Connection to Arches project server
         """
-        # reset connection status on button press
-        self.dlg.connection_status.setText("")
-
-        is_valid_input = True
-        if self.dlg.arches_server_input.text() == "" or str(self.dlg.arches_server_input.text()).isspace() == True:
-            self.dlg.connection_status.append("Please enter the URL to your Arches project.")
-            is_valid_input = False
-        if self.dlg.username_input.text() == "":    
-            self.dlg.connection_status.append("Please enter your username.")
-            is_valid_input = False
-        if self.dlg.password_input.text() == "":
-            self.dlg.connection_status.append("Please enter your password.")
-            is_valid_input = False
+        self.error_msg = ""
+        connection_information = {
+            "URL": {
+                "value": self.dlg.archesServerInput.text().strip(),
+                "input": self.dlg.archesServerInput,
+                "label": self.dlg.archesServerLabel,
+                }, 
+            "username": {
+                "value": self.dlg.usernameInput.text().strip(),
+                "input": self.dlg.usernameInput,
+                "label": self.dlg.usernameLabel
+                }, 
+            "password": {
+                "value": self.dlg.passwordInput.text().strip(),
+                "input": self.dlg.passwordInput,
+                "label": self.dlg.passwordLabel
+                }
+        }
+
+        is_valid_input=True
+        missing_inputs = []
+        for k,v in connection_information.items():
+            if not v["value"]:
+                is_valid_input=False
+                missing_credentials(v["input"], "True")
+                missing_credentials(v["label"], "True")
+                missing_inputs.append(k)
+            else:
+                missing_credentials(v["label"], "False")
+                missing_credentials(v["input"], "False")
 
+        if missing_inputs:
+            if len(missing_inputs) > 1:
+                self.error_msg = f"Login missing values for {'{} and {}.'.format(', '.join(missing_inputs[:-1]), missing_inputs[-1])}"
```

Comment:
This would be more readable without combining and f-string with .format(). I think this should work:
```suggestion
                self.error_msg = f"Login missing values for {', '.join(missing_inputs[:-1])} and {missing_inputs[-1]}."
```